src/BuildingRegistry.Grb.Processor.Upload/UploadProcessor.cs

```diff
@@ -1,20 +1,167 @@
+        private readonly BuildingGrbContext _buildingGrbContext;
+        private readonly ITicketing _ticketing;
+        private readonly IBlobClient _blobClient;
+        private readonly IAmazonECS _amazonEcs;
+        private readonly ILogger<UploadProcessor> _logger;
+        private readonly IHostApplicationLifetime _hostApplicationLifetime;
+        private readonly ECSTaskOptions _ecsTaskOptions;
+
+        public UploadProcessor(
+            BuildingGrbContext buildingGrbContext,
+            ITicketing ticketing,
+            IBlobClient blobClient,
+            IAmazonECS amazonEcs,
+            ILoggerFactory loggerFactory,
+            IHostApplicationLifetime hostApplicationLifetime,
+            IOptions<ECSTaskOptions> ecsTaskOptions)
+        {
+            _buildingGrbContext = buildingGrbContext;
+            _ticketing = ticketing;
+            _blobClient = blobClient;
+            _amazonEcs = amazonEcs;
+            _logger = loggerFactory.CreateLogger<UploadProcessor>();
+            _hostApplicationLifetime = hostApplicationLifetime;
+            _ecsTaskOptions = ecsTaskOptions.Value;
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
         {
             // Check created jobs
-            // See if there's a S3 object with job id
-            // If so, update ticket status and job status => preparing
-            // extract, verify, and store data as job records
-            // update job status => prepared
-            // trigger job processor
-            throw new System.NotImplementedException();
+            var createdJobs = await _buildingGrbContext.Jobs
+                .Where(x => x.Status == JobStatus.Created || x.Status == JobStatus.Preparing)
+                .OrderBy(x => x.Created)
+                .ToListAsync(stoppingToken);
+
+            foreach (var job in createdJobs)
+            {
+                var jobName = new BlobName("received/" + job.BlobName);
+
+                // See if there's a S3 object with job id
+                if (!await _blobClient.BlobExistsAsync(jobName, stoppingToken))
+                {
+                    continue;
+                }
+
+                // If so, update ticket status and job status => preparing
+                await _ticketing.Pending(job.TicketId!.Value, stoppingToken);
+                job.Status = JobStatus.Preparing;
+                await _buildingGrbContext.SaveChangesAsync(stoppingToken);
+
+                var blobObject = await _blobClient.GetBlobAsync(jobName, stoppingToken);
+                if (blobObject is null)
+                {
+                    _logger.LogError($"No blob found with name: {job.BlobName}");
+                    continue;
+                }
+
+                try
+                {
+                    // extract, verify, and store data as job records
+                    await using var stream = await blobObject.OpenAsync(stoppingToken);
```

Comment:
I'd gather all logic related to 
* checking if the blob is present,
* getting the blobObject
* and opening the blob object
in one place.